src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class ILGeneratorImpl : ILGenerator
+    {
+        private const int DefaultSize = 16;
+        private readonly MethodBuilder _methodBuilder;
+        private readonly BlobBuilder _builder;
+        private readonly InstructionEncoder _il;
+        private bool _hasDynamicStackAllocation;
+
+        internal ILGeneratorImpl(MethodBuilder methodBuilder, int size)
+        {
+            _methodBuilder = methodBuilder;
+            _builder = new BlobBuilder(Math.Max(size, DefaultSize));
```

Comment:
Should it throw on size < 1?

Reply:
For compat with the existing runtime ILGenerator:
https://github.com/dotnet/runtime/blob/a590cb4cfb9f1a66c043476695fd0e79835842eb/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeILGenerator.cs#L90

Reply:
That might be worth a comment. It looks like the scenario is covered in a unit test, but the test could use a comment too.